src/Servers/IIS/IIS/test/Common.FunctionalTests/Inprocess/EnvironmentVariableTests.cs

```diff
@@ -126,5 +126,20 @@ private async Task WebConfigExpandsVariables(HostingModel hostingModel)
             deploymentParameters.WebConfigBasedEnvironmentVariables["OtherVariable"] = "%TestVariable%;Hello";
             Assert.Equal("World;Hello", await GetStringAsync(deploymentParameters, "/GetEnvironmentVariable?name=OtherVariable"));
         }
+
+        [ConditionalTheory]
+        [RequiresIIS(IISCapability.PoolEnvironmentVariables)]
+        [InlineData(HostingModel.InProcess)]
+        [InlineData(HostingModel.OutOfProcess)]
+        public async Task PreferEnvironmentVariablesOverWebConfigWhenConfigured(HostingModel hostingModel)
```

Comment:
Oh, we may need the attribute for "RequiresNewShim"/"RequiresNewHandler".

Reply:
It'll be come clear when all tests run.

Reply:
Oh, so split the test like we do for the others above

Reply:
Yeah, so we have regression tests with different attributes which are added based on new functionality. As this is new functionality, some of the regression tests may fail.

Reply:
Fixed via our conversation.